javascript/ql/src/Security/CWE-020/UselessRegExpCharacterEscape.ql

```diff
@@ -0,0 +1,156 @@
+/**
+ * @name Useless regular-expression character escape
+ * @description Prepending a backslash to an ordinary character in a string
+ *              does not have any effect, and may make regular expressions constructed from this string
+ *              behave unexpectedly.
+ * @kind problem
+ * @problem.severity error
+ * @precision high
+ * @id js/useless-regexp-character-escape
+ * @tags correctness
+ *       security
+ *       external/cwe/cwe-20
+ */
+
+import javascript
+import semmle.javascript.CharacterEscapes::CharacterEscapes
+
+newtype TRegExpPatternMistake =
+  /**
+   * A character escape mistake in regular expression string `src`
+   * for the character `char` at `index` in `rawStringNode`, explained
+   * by `mistake`.
+   */
+  TIdentityEscapeInStringMistake(
+    RegExpPatternSource src, string char, string mistake, ASTNode rawStringNode, int index
+  ) {
+    char = getALikelyRegExpPatternMistake(src, mistake, rawStringNode, index)
+  } or
+  /**
+   * An identity-escaped character at `index` of `rawStringNode` that
```

Comment:
This description sounds very general, but the injector only seems to look at `\b` escapes.